benchmark/csharp/Tpch/VectorizedFunctions.cs

```diff
@@ -0,0 +1,121 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using Apache.Arrow;
+
+namespace Tpch
+{
+    internal static class VectorizedFunctions
+    {
+        internal unsafe static DoubleArray ComputeTotal(DoubleArray price, DoubleArray discount, DoubleArray tax)
+        {
+            if (price.Length != discount.Length || price.Length != tax.Length)
+                throw new ArgumentException("Arrays need to be the same length");
+
+            int length = price.Length;
+            var builder = new ArrowBuffer.Builder<double>(length);
+            Vector<double> ones = new Vector<double>(1d);
+
+            fixed (double* pStartExtendedPrice = price.Values)
```

Comment:
Will this run a lot faster than using `GetValue()`? I am thinking whether we should write this in a simpler way for the first version.

Reply:
It actually runs end-to-end in the exact same time whether I use a vectorized implementation, or just add them up one row at a time. I've tried it multiple times and I get the same execution time for query `1v`.

I'll change this to a much more simple approach:

```C#
        internal static DoubleArray ComputeTotal(DoubleArray price, DoubleArray discount, DoubleArray tax)
        {
            if ((price.Length != discount.Length) || (price.Length != tax.Length))
            {
                throw new ArgumentException("Arrays need to be the same length");
            }

            int length = price.Length;
            var builder = new ArrowBuffer.Builder<double>(length);
            ReadOnlySpan<double> prices = price.Values;
            ReadOnlySpan<double> discounts = discount.Values;
            ReadOnlySpan<double> taxes = tax.Values;
            for (int i = 0; i < length; i++)
            {
                builder.Append(prices[i] * (1 - discounts[i]) * (1 + taxes[i]));
            }

            return new DoubleArray(
                builder.Build(),
                nullBitmapBuffer: ArrowBuffer.Empty,
                length: length,
                nullCount: 0,
                offset: 0);
        }
```

Reply:
UPDATE:

I ran query `8v` and I am seeing a slight change in perf:

|                      Query|      Normal UDF|    VectorUdf with `Vector<T>`| VectorUdf with simple `for`|
|---------------------------- |----------:|----------:|----------:|
| Query 8 | 29.7 s | 28.6 s | 29.0 s |

Reply:
I actually like the simple approach since that is how people would normally write (I think).

We can revisit this when we start benchmarking TPCH for the cluster run (along with hardware intrinsics). What do you think?

Reply:
That sounds like a good plan to me.